File: Problem98/Solution.go
import (
	"math"
)
func isValidBST(root *TreeNode) bool {
    if root == nil {
    	return true;
    }
    var MaxInt64  int = 1<<63 - 1
    var MinInt64 int = -1 << 63
    return isValid(root, MaxInt64, MinInt64); 
}

func isValid(p *TreeNode, max int, min int) bool{
	if p == nil {
		return true;
	}
	if p.Val <= min || p.Val >= max {
		return false;
	}
	
	return isValid(p.Left, p.Val, min) && isValid(p.Right, max, p.Val); 
}

File: Problem98/Solution.java
/**
 * @author tgorthi
 * @since Jun 2020
 */
public class Solution
{
    public boolean isValidBST(TreeNode root)
    {
        return isValidBST(root, Long.MIN_VALUE, Long.MAX_VALUE);
    }

    public boolean isValidBST(TreeNode p, Long min, Long max)
    {
        if (p == null)
        {
            return true;
        }
        if (p.val <= min || p.val >= max)
        {
            return false;
        }
        return isValidBST(p.left, min, new Long(p.val)) && isValidBST(p.right, new Long(p.val), max);
    }
}